### PR TITLE
Feature/pdf generation

### PR DESCRIPTION
**Pull request by acrollet**
### Changes made to the platform

This PR adds a platform-level package that will provide templated generation of accessible PDFs. (starting with medical records for the MyHealtheVet on VA.gov project) This meets an immediate business need for MyHealtheVet on VA.gov, but it is anticipated that generation of accessible PDFs is likely to be a need for the whole platform. Package usage looks like this:

```
const getSingleVitalPdf = e => {
  e.preventDefault();

  generatePdf('medical_records', 'vitals', {
    headerLeft: 'Lastname, Firstname',
    headerRight: 'Date of birth: March 15, 1982',
    footerLeft: 'Report generated by My HealtheVet and VA on May 15, 2023',
    footerRight: 'Page %PAGE_NUMBER% of %TOTAL_PAGES%',
    title: 'Blood pressure on April 1, 2022 at 12:34 p.m.',
    preface:
      'Your vitals list may not be complete. If you have any questions about your information, visit the FAQs or contact your VA Health care team.',
    results: {
      items: [
        {
          items: [
            {
              title: 'Result',
              value: '120/80',
              inline: true,
            },
            {
              title: 'Location',
              value: 'Cincinnati VA Medical Center',
              inline: true,
            },
            {
              title: 'Provider notes',
              value: 'None noted',
              inline: true,
            },
          ],
        },
      ],
    },
  });
};

return (
     <div>
        <i
          aria-label="download"
          className="fas fa-download vads-u-color--link-default vads-u-margin-right--1"
          aria-hidden="true"
        />
        <a
          download="singlevital.pdf"
          type="application/pdf"
          href="singlevital.pdf"
          onClick={getSingleVitalPdf}
        >
          Download Vitals
        </a>
      </div>
);
```

### Why is this the solution

We initially looked into generating these PDFs on the server-side, but ran into the following issues:
- There are no Ruby packages available that can generate PDFs _de novo_ with semantic markup for accessibility (Existing accessible PDFs generated by vets-api are either inaccessible or created by filling form fields in a hand-generated PDF template)}
- There is no existing path to serve node.js applications to VA.gov

More background is available in [this slack thread](https://dsva.slack.com/archives/C03Q2UQL1AS/p1680636566262409)

### Team

I'm an OCTO engineer working on the health applications team. We're happy to share the burden of maintenance on this package.

## Related issue(s)

- Accessibility review ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/58587

## Testing done

Unit tests are present to confirm generation functionality and semantic markup.

## Screenshots

This package does not provide its own UI.

## What areas of the site does it impact?

No existing applications are modified by this package.

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated (in package README)
- [ ] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by acrollet** [comment]
I've got a PR [here](https://github.com/department-of-veterans-affairs/vets-website/pull/24479) for the unit test that's failing. ([related slack thread](https://dsva.slack.com/archives/C03ECSBGSKX/p1686167413591189))

**Comment by acrollet** [comment]
Here's a link to a review environment with a demo: http://0c9efd4e92fd802908e2521a9bd48a00.review.vetsgov-internal/health-care/appointment-pre-check-in/error?error=no-token (from [this separate PR](https://github.com/department-of-veterans-affairs/vets-website/pull/24059))

**Comment by acrollet** [comment]
> Looks like there's a unit test failure when run after `src/platform/forms-system/test/js/utilities/file.unit.spec.js`, looking into it

fixed in this PR: https://github.com/department-of-veterans-affairs/vets-website/pull/24494

File: src/platform/pdf/templates/medical_records.js [hunk 1]
```diff
@@ -0,0 +1,354 @@
+/**
+ * Lab and Test Results PDF template.
+ *
+ * NB: The order in which items are added to the document is important,
+ * and thus PDFKit requires performing operations synchronously.
+ */
+/* eslint-disable no-await-in-loop */
```
[the remark this answers is not in the document]
Comment (acrollet) [reply src/platform/pdf/templates/medical_records.js:7]: Disabled because document sections must be added in order.

**Review by rsmithadhoc — APPROVED** [review]

**Review by mdewey — APPROVED** [review]
LGTM! I am excited about where this will go!

**Review by stephenBarrs — APPROVED** [review]
LGTM!

**Pull request merged**